src/entity.cpp

```diff
@@ -3699,7 +3699,23 @@ void Entity::handleEffects(Stat* myStats)
 				{
 					if ( myStats->shield->count > 1 )
 					{
-						newItem(myStats->shield->type, myStats->shield->status, myStats->shield->beatitude, myStats->shield->count - 1, myStats->shield->appearance, myStats->shield->identified, &myStats->inventory);
+						Uint32 newAppearance = rand();
+						int tries = 100;
+						while ( tries > 0 && newAppearance == myStats->shield->appearance )
+						{
+							--tries;
```

Comment:
this decrements `tries` but doesn't `newAppearance = rand();` also maybe it would be better to just:
```
newAppearance = rand();
if (newAppearance == myStats->shield->appearance)
    newAppearance++;
```

?

Reply:
I wasn't happy with this solution, reverted the appearance change. This pattern appears a lot across the code and I'd rather look at it at some other stage